src/app/phase2/issues-faulty/issues-faulty.component.ts

```diff
@@ -32,6 +36,17 @@ export class IssuesFaultyComponent implements OnInit, OnChanges {
       this.displayedColumns = ['id', 'title', 'teamAssigned', 'type', 'severity', 'responseTag', 'assignees', 'duplicatedIssues',
         'actions'];
     }
+
+    this.navigationSubscription = this.router.events.subscribe((e: any) => {
+      // If it is a NavigationEnd event re-initalise the data
+      if (e instanceof NavigationEnd) {
+        if (this.runOnce) {
+          this.issueService.reset();
+          this.initialiseData();
+          this.table.renderRows();
+        }
+      }
```

Comment:
Can we use a guard statement to reduce nesting? e.g.,
```js
if !(e instanceof NavigationEnd) {
    return;
}
...
```

Reply:
Noted, I have removed the nested if condition.